### DISCUSSION TITLE
vipsthumbnail *.jpg error

### DISCUSSION THREAD

**Discussion by zc2tech** [libvips/libvips#2091] [Q&A]
**Describe the bug**
I cannot sue this command:
vipsthumbnail *.jpg

**To Reproduce**
Version: vips-dev-8.10

F:\picture\>vipsthumbnail *.jpg --size "1920x1920>"
vipsthumbnail: unable to thumbnail *.jpg
VipsForeignLoad: file "*.jpg" does not exist

**Comment by jcupitt** [comment]
Hi @johnsonshu,

Sorry, I don't know. Perhaps there are no `.jpg` files in that directory?

Try setting `--vips-info`, eg.:

```
vipsthumbnail *.jpg --size "1920x1920>" --vips-info
```

It might give a hint about what's happening.

**Reply by powerlicht** [reply]
FAIL in Windows-Console:
 `vipsthumbnail *.jpg --size "1920x1920>" --vips-info`

GOOD in Windows-Console:
`FOR %f IN (*.jpg) DO vipsthumbnail "%f" --size "1920x1920>" --vips-info`

**Reply by jcupitt** [reply]
Exactly, windows doesn't have filename globbing in the shell, you need to loop.

**Reply by powerlicht** [reply]
Nice. I learned a new word - globbing - to maybe complete this, if another Windows user stumbled across this, then maybe the variant from a **batch-file**
`FOR %%f IN (*.jpg) DO @(vipsthumbnail.exe "%%f" --size "1920x1920>" --vips-info)`

**Comment by jcupitt** [comment]
(I changed "bug" to "question" because I don't think this is a libvips issue)

**Comment by Transigent** [comment]
Yes I had this problem as well. 

I found in the comments section here: http://libvips.blogspot.com/2013/11/tips-and-tricks-for-vipsthumbnail.html
that apparently Windows will not do 'globbing' and wont decode *.jpg like other systems. Somebody suggested using powershell but I had no success. 

In the end I used this:
  `vipsthumbnail file1.jpg file2.jpg file3.jpg --size=640 -o tn_%s.webp[strip] --vips-progress`

Note also that this would not work for me in a batch file, I would end up with only one thumb file (the last one processed). I had to paste it in and it worked properly then.

**Comment by jcupitt** [comment]
Ah you're right @Transigent, I'd forgotten that Windows cmd.exe doesn't really support wildcards in filenames. Yes, powershell is probably the best answer.
